neo4j-jdbc/src/main/java/org/neo4j/jdbc/ConnectionImpl.java

```diff
@@ -661,6 +691,42 @@ public void flushTranslationCache() {
 		}
 	}
 
+	private <T extends AutoCloseable> T addAutoCloseable(T autoCloseable) {
+		this.instantiatedAutoClosables.add(autoCloseable);
+		return autoCloseable;
+	}
+
+	@Override
+	public void onClose(AutoCloseable autoCloseable) {
+		this.instantiatedAutoClosables.remove(autoCloseable);
+	}
+
+	private void handleFatalException(SQLException fatalSqlException, SQLException sqlException) {
+		var cause = sqlException.getCause();
+		if (cause instanceof ConnectionReadTimeoutException) {
+			var iterator = this.instantiatedAutoClosables.iterator();
+			while (iterator.hasNext()) {
+				var autoClosable = iterator.next();
+				iterator.remove();
+				try {
```

Comment:
Can pull the next into to the `try()`and do remove in the finally block. Feels better.

Reply:
Done. 👍

Reply:
Updated as agreed.